### ISSUE TITLE
[Feature]  File Manager Improvements

### ISSUE THREAD

**Issue by Digioso** [hestiacp/hestiacp#4176]
### Describe the feature or change in detail

Hi!

Today I did a few things in the file manager and noticed some areas that should be improved.
I'm using Hestia CP v1.8.10.
a) Error messages. The FM doesn't give any error messages whatsoever.
Example: Try to create a new folder in the home root directory of a user.
The new folder is not there, but also no error message is given.
According to [this post ](https://forum.hestiacp.com/t/file-manager-root-folder/11746) you cannot create stuff in the root directory. That's fine, but an error message when trying to do so would be very nice.
b) Unzip function doesn't overwrite existing files.
A new version of a CMS I'm using was released yesterday. As with many web applications you download an upgrade package, upload it to your site, extract it and overwrite the existing files.
Only thing is... the unzip function in the file manager does not overwrite existing files. You click on unzip and then... nothing. No error messages is given, but the files are not extracted either.
Expected behaviour would be that it asks the user whether they want to overwrite existing files or not.
I did a few tests and can confirm this happens with any zip file.

**Comment by maisondasilva** [comment]
The file manager used is filegator as @jaapmarcus commented in another post for me, I believe you can open an issue in the repository filegator!

https://github.com/filegator/filegator

https://github.com/filegator/filegator/issues

**Comment by Digioso** [comment]
Ah, thanks! I didn't know that.
It looks like someone has already requested the overwrite feature: https://github.com/filegator/filegator/issues/390
And for the error messages I added a new post to an existing issue as well: https://github.com/filegator/filegator/issues/261

**Comment by jaapmarcus** [comment]
I sure we use for unzipping our own function as it is a lot faster...

https://github.com/hestiacp/hestiacp/blob/main/install/deb/filemanager/filegator/backend/Services/Archiver/Adapters/HestiaZipArchiver.php

We might be able to improve that part...

**Comment by vdbhb59** [comment]
While I commented with my issues on filegator, is there a possibility of moving to some other file manager?

**Comment by virtualos** [comment]
Ok, I want to update my CMS. I uploaded new zip file through Hestia file manager and tried to unzip and after an hour of confusion I got that there is no overwrite function!!!!
This must be fixed asap guys! this is a very important fix!!

Does anyone have any patch I can use now?

**Comment by jaapmarcus** [comment]
You should open an issue issue at Filegator instead for it..

**Comment by vdbhb59** [comment]
> You should open an issue issue at Filegator instead for it..

But like I mentioned two posts above, can we not look at a different file manager? Example, like the one cPanel uses, which is really very efficient.

**Comment by virtualos** [comment]
I will open issue at filegator but I am agree with @vdbhb59 can you use another more mature file explorer for hestiacp?

**Comment by Digioso** [comment]
> I will open issue at filegator but I am agree with @vdbhb59 can you use another more mature file explorer for hestiacp?

I second that!

**Comment by vdbhb59** [comment]
I located a couple of other file managers. Not sure if those can be used, or how those can be integrated. Hope someone can assist. 🙂 
* [File Browser](https://github.com/filebrowser/filebrowser) -- Very simple approach in terms of UI, and nicely done.
* [File Stash](https://github.com/mickael-kerjean/filestash) -- Very heavy I believe, but still a good one.

**Comment by jaapmarcus** [comment]
> I located a couple of other file managers. Not sure if those can be used, or how those can be integrated. Hope someone can assist. 🙂
> 
> * [File Browser](https://github.com/filebrowser/filebrowser) -- Very simple approach in terms of UI, and nicely done.

Lacks sftp support

> * [File Stash](https://github.com/mickael-kerjean/filestash) -- Very heavy I believe, but still a good one.

Runs on Go  so we need to start a second server. Really don't want that ...

**Comment by vdbhb59** [comment]
> Lacks sftp support

> Runs on Go so we need to start a second server. Really don't want that ...

Yeah, I thought so. I was not sure if it will work or not, but am crawling other git hosting sites to hopefully find something.

**Comment by rjd22** [comment]
It would be better to just update fileGator IMO. if you look at the code it's a more mature and easily maintainable library.

I feel changing file managers is throwing out the baby with the bathwater.

**Comment by vdbhb59** [comment]
I agree. I was just trying to think of a way, as everyone is facing a lot of issues with them. + it is limited in ways. But that is okay I guess.
